activerecord/lib/active_record/relation/batches.rb

```diff
@@ -274,46 +304,60 @@ def in_batches(of: 1000, start: nil, finish: nil, load: false, error_on_ignore:
           end
         end
 
-        batch_relation = batch_condition(
-          relation, primary_key, primary_key_offset, order == :desc ? :lt : :gt
-        )
+        batch_order_array = batch_order.to_a
+        last_column, last_order = batch_order_array.pop
+        operators = {}
+        batch_order_array.map do |column, order|
+          operators[column] = order == :desc ? :lteq : :gteq
+        end
+        operators[last_column] = last_order == :desc ? :lt : :gt
```

Comment:
Despite of ruby will enumerate over `Hash` in order of keys insertion I would prefer `operators` to stay as an array as on inspection it will be clearer which element exactly is the last while it may not be exactly clear if we use a hash.

Sorry, I might have said "hash" explicitly in my previous comments but ideally I think the data should be represented by an array

Reply:
@nvasilevski
I did this because  [hash based api might be implemented in the future](https://github.com/rails/rails/pull/48268#pullrequestreview-1440030742), but I did not need to do it now, and you are right that an array is easy to understand the order. I fixed it, so please check it out.